### DISCUSSION TITLE
Should we keep this project as separate lib/bin crates and should we publish to crates.io?

### DISCUSSION THREAD

**Discussion by jhodapp** [Jim-Hodapp-Coaching/ambi_mock_client#14] [General]
@AntoniosBarotsis proposed the idea of publishing ambi_mock_client to crates.io for ease of use (i.e. `cargo install ambi_mock_client` is then possible). To do that, we'd want to collapse back down to a single bin crate from our existing separate lib/bin crates in a single project. What do other project members think of this idea...open for discussion.

Attn: @calebbourg, @dilyn-corner, @naconnors, @qafui

**Comment by AntoniosBarotsis** [comment]
It is worth noting that we could also make it a proper installable CLI without publishing it to crates.io, we could install it from path instead which I think is more convenient than what we have now.

**Reply by calebbourg** [reply]
@AntoniosBarotsis I think either options are valid. Maybe we go with this option first and decide if it makes sense to publish to crates.io somewhere down the line? I do like the idea of being able to just use `cargo install`

**Reply by jhodapp** [reply]
Agreed Caleb, similar to what we did with the `esp32-wroom-rp` crate, we can build up and graduate to publishing to crates.io if/when we feel it's appropriate. As of right now I see this mainly as a tool for Ambi developers who don't have the full Raspberry Pi Pico and sensors hardware to use for sourcing sensor values.
